File: version4j/src/main/java/com/only4/algorithm/version4j/leetcode/InvertTree.java
package com.only4.algorithm.version4j.leetcode;

import com.only4.algorithm.version4j.extra.TreeNode;

public class InvertTree {
    public TreeNode invertTree(TreeNode root) {
        if (root == null) return null;

        TreeNode leftSubtree = invertTree(root.left);
        TreeNode rightSubtree = invertTree(root.right);

        root.left = rightSubtree;
        root.right = leftSubtree;

        return root;
    }
}

File: version4k/src/main/kotlin/com/only4/algorithm/version4k/leetcode/invertTree.kt
package com.only4.algorithm.version4k.leetcode

import com.only4.algorithm.version4k.extra.TreeNode

fun invertTree(root: TreeNode?): TreeNode? {
    if (root == null) return null

    val leftSubtree = invertTree(root.left)
    val rightSubtree = invertTree(root.right)

    root.left = rightSubtree
    root.right = leftSubtree

    return root
}
